pkgs/applications/editors/vscode/extensions/vscode-utils.nix

```diff
@@ -9,38 +9,36 @@
   jq,
 }:
 let
-  buildVscodeExtension =
-    a@{
-      pname ? null, # Only optional for backward compatibility.
-      src,
-      # Same as "Unique Identifier" on the extension's web page.
-      # For the moment, only serve as unique extension dir.
-      vscodeExtPublisher,
-      vscodeExtName,
-      vscodeExtUniqueId,
-      configurePhase ? ''
-        runHook preConfigure
-        runHook postConfigure
-      '',
-      buildPhase ? ''
-        runHook preBuild
-        runHook postBuild
-      '',
-      dontPatchELF ? true,
-      dontStrip ? true,
-      nativeBuildInputs ? [ ],
-      passthru ? { },
-      ...
-    }:
-    stdenv.mkDerivation (
-      (removeAttrs a [
-        "vscodeExtUniqueId"
-        "pname"
-      ])
-      // (lib.optionalAttrs (pname != null) {
+  buildVscodeExtension = lib.extendMkDerivation {
+    constructDrv = stdenv.mkDerivation;
+    excludeDrvArgNames = [
+      "vscodeExtUniqueId"
+    ];
+    extendDrvArgs =
+      finalAttrs:
+      {
```

Comment:
should we keep `src` in here? even if it is not used later?

Reply:
If I would like to remove it from the set pattern, I would remove it in separate commits. It would be easier to bisect should something unexpected happen.

Reply:
OK fair enough, I have added it back and made a new commit removing it.

Reply:
As for whether to remove `src` from the set pattern, I don't know if people have reached consensus on it. Some (once heated) discussion can be found in issue #331085.

Reply:
IMO, it makes sense here since we are introducing a new builder. Ideally, this builder should only include relevant attributes (e.g., `vscodeExtPublisher`, ...). Since `src` is required by `stdenv.mkDerivation` **and** we are using `...`, keeping `src` here might cause confusion. Unless I am missing something, it can be removed safely.